packages/eslint-plugin-pf-codemods/src/rules/v6/formFiledGroupHeaderTitleTextObjectRenamed/formFiledGroupHeaderTitleTextObject-renamed.ts

```diff
@@ -0,0 +1,59 @@
+import { AST, Rule } from "eslint";
+import { ImportSpecifier, Identifier, Node } from "estree-jsx";
+import { getFromPackage } from "../../helpers";
+
+// https://github.com/patternfly/patternfly-react/pull/10016
+module.exports = {
+  meta: { fixable: "code" },
+  create: function (context: Rule.RuleContext) {
+    const { imports } = getFromPackage(context, "@patternfly/react-core");
+
+    const previousName = "FormFiledGroupHeaderTitleTextObject";
+    const newName = "FormFieldGroupHeaderTitleTextObject";
+
+    const interfaceImport = imports.find(
+      (specifier) => specifier.imported.name === previousName
+    );
+
+    if (!interfaceImport) {
+      return {};
+    }
+
+    const hasAlias = (specifier: ImportSpecifier) =>
+      specifier.local.name !== specifier.imported.name;
+
+    const reportMessage =
+      "There was a typo in FormFiledGroupHeaderTitleTextObject interface. It was renamed to the intended FormFieldGroupHeaderTitleTextObject.";
+
+    const callContextReport = (node: Node, nodeToReplace: Node | AST.Token) => {
+      context.report({
+        node,
+        message: reportMessage,
+        fix(fixer) {
+          return fixer.replaceText(nodeToReplace, newName);
+        },
+      });
+    };
+
+    return {
+      ImportSpecifier(node: ImportSpecifier) {
+        if (node.imported.name === interfaceImport.imported.name) {
+          callContextReport(node, node.imported);
+        }
+      },
+      TSTypeReference(node: { typeName: Identifier }) {
+        if (!hasAlias(interfaceImport) && node.typeName.name === previousName) {
+          callContextReport(node as unknown as Node, node.typeName);
+        }
+      },
+      TSInterfaceHeritage(node: { expression: Identifier }) {
```

Comment:
Having these typed as objects with just the expected values and then asserting that they're Nodes feels odd, are there not better types we could use here?

Reply:
Yeah, I know. I was struggling to find any type for these, there are only enums for these in `/packages/eslint-plugin-pf-codemods/lib/rules/ast-node-types.d.ts` which is I think is copied from `@typescript-eslint/parser`. But I am taking a look at the typescript-eslint repo now and it seems the actual interface might be exported there.

I'll try to import that

Reply:
Hm 🤔 

Let me do a little hunting myself and see if I can find the types we need exported anywhere.

Reply:
I think I found them:

`import { TSTypeReference, TSInterfaceHeritage} from '@typescript-eslint/parser'`

Reply:
I tried importing this, the '@typescript-eslint/parser' package should be installed, but it says it is not exported.

![Screenshot 2024-03-25 at 11 27 55](https://github.com/patternfly/pf-codemods/assets/84135613/21dbddf4-7811-435b-8884-10578d800838)

Updated `packages/eslint-plugin-pf-codemods/package.json`:

![Screenshot 2024-03-25 at 11 27 04](https://github.com/patternfly/pf-codemods/assets/84135613/add6ec30-d89b-47ae-9d36-f85357037512)

Then I did `yarn install`
I don't know if I am installing something wrong, but it just seems like it is not exported.

Reply:
I'm seeing the same thing but I swear this was working on Friday 🤔

Reply:
I really am sure that this was working Friday and now I'm so confused 😆 

It looks like types of those names can be retrieved from `import { TSESTree } from '@typescript-eslint/types'
`... but they don't slide right into place and I'm not sure if those are the wrong type or if the current logic isn't fully type safe.

I do feel like we should investigate this further but I won't block this PR over it if you'd like to make a followup issue so that we don't forget about it.

Reply:
Typescript contributors are playing some high-level games with us I guess 😃 

Anyways, the `TSESTree.AST_NODE_TYPES` enum si fine, but it does not fit the use case as you say. I'd also not block over, the codemod should work even if typed as `TSTypeReference(node: any) {`, as if the node matches that TSTypeReference function, it should already have the suitable structure I suppose.

Opened the issue for further investigation: https://github.com/patternfly/pf-codemods/issues/622

Reply:
Yeah the codemod will still work, which is why I'm fine with the typing being patched in a followup. It just isn't ideal from a typing perspective since it isn't _really_ valid, we're just telling TS to trust us and chill out 😆